taca/analysis/analysis.py

```diff
@@ -308,132 +283,41 @@ def samplesheet_to_dict(samplesheet):
-        adapter_stringency = cl_options.get('adapter-stringency')
-        if adapter_stringency and type(adapter_stringency) is float:
-            cl.extend(['--adapter-stringency', adapter_stringency])
-        aggregated_tiles = cl_options.get('aggregated-tiles')
-        if aggregated_tiles and aggregated_tiles in ['AUTO', 'YES', 'NO']:
-            cl.etend(['--aggregated-tiles', aggregated_tiles])
-        barcode_missmatches = cl_options.get('barcode-missmatches')
-        if barcode_missmatches and type(barcode_missmatches) is int \
-                and barcode_missmatches in range(3):
-            cl.extend(['--barcode-missmatches', barcode_missmatches])
-        if cl_options.get('create-fastq-for-index-reads'):
-            cl.append('--create-fastq-for-index-reads')
-        if cl_options.get('ignore-missing-bcls'):
-            cl.append('--ignore-missing-bcls')
-        if cl_options.get('ignore-missing-filter'):
-            cl.append('--ignore-missing-filter')
-        if cl_options.get('ignore-missing-locs'):
-            cl.append('--ignore-missing-locs')
-        mask = cl_options.get('mask-short-adapter-reads')
-        if mask and type(mask) is int:
-            cl.extend(['--mask-short-adapter-reads', mask])
-        minimum = cl_options.get('minimum-trimmed-reads')
-        if minimum and type(minimum) is int:
-            cl.extend(['--minimum-trimmed-reads', minimum])
-        if cl_options.get('tiles'):
-            cl.extend(['--tiles', cl_options.get('tiles')])
-
-        if cl_options.get('use-bases-mask'):
-            cl.extend(['--use-bases-mask', cl_options.get('use-bases-mask')])
-
-        if cl_options.get('with-failed-reads'):
-            cl.append('--with-failed-reads')
-        if cl_options.get('write-fastq-reverse-complement'):
-            cl.append('--write-fastq-reverse-complement')
-
-        LOG.info(("BCL to FASTQ conversion and demultiplexing started for "
-                  " run {} on {}".format(os.path.basename(run), datetime.now())))
-
-        misc.call_external_command_detached(cl, with_log_files=True)
-
-        LOG.info(("BCL to FASTQ conversion and demultiplexing finished for "
-                  "run {} on {}".format(os.path.basename(run), datetime.now())))
-        # Transfer the processed data to the analysis server
-        transfer_run(run, config)
-
-
-def run_demultiplexing(run):
+def run_preprocessing(run):
     """ Run demultiplexing in all data directories """
+
     config = CONFIG['preprocessing']
 
     hiseq_runs = glob.glob(os.path.join(config['hiseq_data'], '1*XX')) if not run else [run]
-    for run in hiseq_runs:
-        run_name = os.path.basename(run)
-        LOG.info('Checking run {}'.format(run_name))
-        if is_finished(run):
-            status = processing_status(run)
-            if  status == 'TO_START':
+    for _run in hiseq_runs:
+        run = Run(_run)
+        LOG.info('Checking run {}'.format(run.id))
+        if run.is_finished():
```

Comment:
the run.status attribute is used for the _demultiplexing_ status at the moment, so `FINISHED` means that the demultiplexing is dome. `is_finished` tells you if the sequencing is done, which is what we want here. Does it makes sense?